### DISCUSSION TITLE
ESPHome local web files ?

### DISCUSSION THREAD

**Discussion by MattWestb** [tube0013/tube_gateways#115] [Ideas]
ESPHome default using all web contents from ESPHome servers.
That is good then no need saving them in the flash.
But if the servers is not online then its one outage or problem with ISP or the local network the ESPHome dashboard is not working in the host system and also all devices cant shoeing there WEB GUI.
They was one outage for for some months ago and and it was not possible doing ny thing with my ESPH devices :-(((

Was looking in the ESPH docks and its shall being possible string all WEB contents in the device flash.
https://esphome.io/components/web_server.html#here-be-dragons

Have trying it on ESP8266 but was not getting it working OK.

I think it can being good for user with ESP32 working also if the internet connection is not working OK so can diagnose the system .

More thinking and testing is more then welcome !!!

**Comment by tube0013** [comment]
yes this has been on my list of things to try.

Right now my focus has been optimizing the current code for speed/stability. and also confirming the code works with the IDF framework - which in initial testing it is working very well. IDF brings along other issues though including no web frontend - but there is a PR that works!

**Reply by MattWestb** [reply]
Good that is you is aware of that no internet (user have blocking the device for getting internet in the router) is making the WEB GUI not working on the device.

I keep looking whats happens and if somthing working id coming out of the 2023 pipe :-)))
